x/config/chooser.go

```diff
@@ -52,6 +52,22 @@ func (pc PeerListConfig) Empty() bool {
 	return c.Peer == "" && len(c.Etc) == 0
 }
 
+// Peer returns the configured peer or returns an error if there is no peer
+// configured, or if there are extra config attributes.
+// This is a facility for the TChannel ChannelTransport peer configuration
+// specifically since it is unable to make use of any other configuration
+// properties.
```

Comment:
Seeing as this is a public API, probably shouldn't mention that this is for the TChannel ChannelTransport. Maybe add that to a comment.